### PR TITLE
Remove v2.3 migration code

### PR DESCRIPTION
**Pull request by prateekdefi**
Operational steps prior to merge:
- [ ] Ensure all local positions are refreshed with the new v2.3 storage layout
- [ ] Claim the idle DSU in the `OracleFactory`

### CODE REVIEW

**Review by EdNoepel — APPROVED** [review]
LGTM

**Comment by kbrizzle** [comment]
There's a few more spots (mostly in the oracle) - try grepping for `v2.3` in the solidity files.

Also there's two comments here which require a full user settlement or state refresh (trigger orders and position). Let's mark a TODO on this PR that those processes must be complete before we merge this in. cc: @arjun-io.

**Comment by kbrizzle** [comment]
Random q - does code size "change" not work anymore? I saw that working on a prior PR, but this one says no change which doesn't seem right.

**Comment by EdNoepel** [comment]
> > Random q - does code size "change" not work anymore? I saw that working on a prior PR, but this one says no change which doesn't seem right.
> 
> It will work after #488 PR gets merge. @kbrizzle
Background: He had to merge a CI step to generate report in the development branch (v2.4) such that PR #488 has something to compare to.  But until that is merged, every report is expected to show 0 diff.

**Comment by kbrizzle** [comment]
The `KeeperOracle_Migration` and associated contracts can be fully removed as well.

File: packages/periphery/contracts/MultiInvoker/types/TriggerOrder.sol [hunk 1]
```diff
@@ -34,7 +34,7 @@ struct StoredTriggerOrder {
     /* slot 2 */
     address interfaceFeeReceiver2;
     uint48 interfaceFeeAmount2;      // <= 281m
-    bytes6 __unallocated2__;         // Contains dirty data until updated post v2.3 migration.
+    bytes6 __unallocated2__;
```
Comment (kbrizzle) [inline packages/periphery/contracts/MultiInvoker/types/TriggerOrder.sol:37]: nit: we still need this comment as well.

Comment (prateekdefi) [reply packages/periphery/contracts/MultiInvoker/types/TriggerOrder.sol]: Fixed in [e87ed90](https://github.com/equilibria-xyz/perennial-v2/pull/496/commits/e87ed90f7ead7ee150e7307a0fe1892f4b12eca4)

**Review by kbrizzle — APPROVED** [review]

**Pull request merged**